packages/kbn-unified-doc-viewer/src/services/doc_views_registry.ts

```diff
@@ -50,8 +56,30 @@ export class DocViewsRegistry {
     this.docViews.delete(id);
   }
 
+  enableById(id: string) {
```

Comment:
Agree, ideally we could initialize a DocViewsRegistry instance for a specific flavoured version of discover and set its doc views, this would remove the need for cloning the registry as every version would have its own instance, or at list the control of the registries would be centralized in discover. We can definitely tackle this once the code-base is more unified, meanwhile, I improved the test of clone to assert against mutations.